quran/test_fields.py

```diff
@@ -0,0 +1,216 @@
+import shutil
+import tempfile
+from django.db import connection
+from django.core import exceptions
+from django.core.files import File
+from django.conf import settings
+from django.test import TestCase, SimpleTestCase
+from .models import Ayah, Reciter, Recitation
+from .fields import SegmentsField
+
+
+class SegmentsFieldTestCase(TestCase):
+    '''Test for SegmentsField'''
+
+    def setUp(self):
+        # Create a temporary directory
+        self.test_dir = tempfile.mkdtemp()
+        self._original_media_root = settings.MEDIA_ROOT
+        settings.MEDIA_ROOT = self.test_dir
+        # get the 93rd ayah from the surah Al-Baqara
+        self.ayah_2_93 = Ayah.objects.get(pk=100)
+        # create a temporary audio
+        file = tempfile.NamedTemporaryFile(suffix='.mp3')
+        self.audio_ayah = File(file, name=file.name)
+        # create a reciter
+        self.husari = Reciter.objects.create(
```

Comment:
Just by looking at this line I understood that your tests are wrong. You don't need any models to test your fields. Read few tests how other django fields are tested: https://github.com/django/django/tree/master/tests/model_fields

What you really need to test is your few methods converting from strings to python objects and vice versa.